File: vertx-lang-kotlin-coroutine/src/main/java/io/vertx/kotlin/coroutines/AbstractVerticle.java
package io.vertx.kotlin.coroutines;

import io.vertx.core.Context;
import io.vertx.core.Verticle;
import io.vertx.core.Vertx;

/**
 * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
 */
public abstract class AbstractVerticle implements Verticle {

  protected Vertx vertx;
  protected Context context;

  @Override
  public Vertx getVertx() {
    return vertx;
  }

  @Override
  public void init(Vertx vertx, Context context) {
    this.vertx = vertx;
    this.context = context;
  }
}

File: vertx-lang-kotlin-coroutine/src/main/kotlin/io/vertx/kotlin/coroutines/AbstractVerticle.kt
package io.vertx.kotlin.coroutines

import io.vertx.core.Context
import io.vertx.core.Verticle
import io.vertx.core.Vertx

/**
 * @author [Julien Viet](mailto:julien@julienviet.com)
 */
abstract class AbstractVerticle : Verticle {

    private lateinit var vertxInstance: Vertx
    protected lateinit var context: Context

    override fun init(vertx: Vertx, context: Context) {
        this.vertxInstance = vertx
        this.context = context
    }

    override fun getVertx(): Vertx = vertxInstance
}
